src/basic/path-util.c

```diff
@@ -336,6 +337,8 @@ char *path_simplify(char *path, PathSimplifyFlags flags) {
          * ///foo//./bar/.   becomes /foo/bar          (if PATH_SIMPLIFY_KILL_DOT is specified)
          * .//./foo//./bar/. becomes ././foo/./bar/.   (if PATH_SIMPLIFY_KILL_DOT is not specified)
          * .//./foo//./bar/. becomes foo/bar           (if PATH_SIMPLIFY_KILL_DOT is specified)
+         * .//./foo/../bar/. becomes ././bar           (if PATH_SIMPLIFY_KILL_DOUBLE_DOT is specified)
+         * .//./foo/../bar/. becomes bar               (if PATH_SIMPLIFY_KILL_DOT and …_DOUBLE_DOT are specified)
```

Comment:
uhhh, this is not ok in the general case.

If "foo" is a symlink to "quux/piff" then "foo/../bar" resolves to "quux/bar" quite likely, and not to "/bar"...

What's the usecase for this? We really shouldn't do this for the general case.

Reply:
Duh, I messed this up.